### ISSUE TITLE
assignment of the name for a session

### ISSUE THREAD

**Issue by PeterGilles** [IPS-LMU/emuR#207]
I am creating an emuDB from TextGrids:

```
# get the path to a folder containing .wav & .TextGrid files that is part of the demo data
textgrids = file.path("/Users/xxxx/Documents/xxx/sound_files", "75_Fraesch", "Wav+TextGrids")

# convert this TextGridCollection to the emuDB format
convert_TextGridCollection(textgrids, dbName = "fraesch2",
                           targetDir = tempdir())
```
In the resulting emuDB-Folder I then have a directory like `Users/xxxx/Documents/xxx/sound_files/75_Fraesch/Wav+TextGrids_ses` instead of `0000_ses'. How is the name for `*_ses' assigned?

**Comment by raphywink** [comment]
That is due to `convert_TextGridCollection()` allowing for subfolder stuctures to be preserved in the session names. `?convert_TextGridCollection()` states about it's `dir` parameter:

- `dir`: path to directory containing the TextGridCollection (nested directory structures are permitted as the function recursively searches through directories, generating the session names from dir. structure)

In other words if your have following folder structure:

- dir/subdir1/subsubdir1
- dir/subdir2
- dir

and every folder contains some TextGrid & wav file pair, `convert_TextGridCollection()` will preserve that folder structure in it's session structure:

- `0000_ses` containing the dir file pairs
- `subdir1_ses` containig the subdir1 file pairs
- `subdir1_subsubdir1_ses` ...
- `subdir2_ses` ...

**Comment by PeterGilles** [comment]
Hm, maybe I am not really understanding the structure. When I run the commands from the tutorial:

```# get the path to a folder containing .wav & .TextGrid files that is part of the demo data
path2folder = file.path(tempdir(), "emuR_demoData", "TextGrid_collection")

# convert this TextGridCollection to the emuDB format
convert_TextGridCollection(path2folder, dbName = "myTGcolDB", 
```

... I will have the bundles stored in `0000_ses`.

When I try the same with my data:

```# convert this TextGridCollection to the emuDB format
convert_TextGridCollection("/var/folders/jc/kd0qwx_1409523wqzlmp03xr5k9m2z/T//RtmpZIGTOz/emuR_demoData/Wav+TextGrids", dbName = "fraesch",
                           targetDir = tempdir())
```

... I am only getting a these files in the _emuDB directory:

```fraesch_DBconfig.json
private_var_folders_jc_kd0qwx_1409523wqzlmp03xr5k9m2z_T_RtmpZIGTOz_emuR_demoData_Wav+TextGrids_ses
```

EMU-WebApp then is also not accepting this long name.

**Comment by raphywink** [comment]
That is super strange. Have you tried maybe using a different folder than tempdir() just something for a quick test like your Desktop? Just out of interesst. What OS are you running and the output of devtools::session_info() might also help me debug the issue.

**Comment by PeterGilles** [comment]
I often encounter strange errors ;) I am on OSX. I tried with a different directory:

```
dir = "/Users/xxx.xxx/Documents/Desktop"

# get the path to a folder containing .wav & .TextGrid files that is part of the demo data
textgrids = file.path(dir, "75_Fraesch", "Wav+TextGrids")

# convert this TextGridCollection to the emuDB format
convert_TextGridCollection(textgrids, dbName = "fraesch",
                           targetDir = dir)

```
Result:
```
Desktop xxx.xxx$ ls fraesch_emuDB/
Users_xxx.xxx_Desktop_Wav+TextGrids_ses
fraesch_DBconfig.json
```

However, when I copy all my wav and textgrid files to the demo folder `emuR_demoData/TextGrid_collection` I will have a `0000_ses` folder.

My aim is to MAUS align many wave files of prompted speech for which I have simple BPF transcriptions. Since the language is not supported, I am using the cURL service and `sampa` as language in MAUS. Is there a way to directly import and annotate the file directly in emuDB?

**Comment by raphywink** [comment]
hmm... very strange. Just checked all operating systems that are easily available to me (OSX, Windows 10 and Ubuntu) and it seems to work on all of them. Could you maybe send me a minimal example incl. data + script so I can reproduce the problem? Somehow `convert_TextGridCollection` seems to be concatenating the passed in path variable to an underscored version... which is super strange. FYI: Can also email it to me if you don't want to have the data up on GitHub.

**Comment by PeterGilles** [comment]
Finally, I think I've found the culprit: The folder containing the files was named `Wav+TextGrids`. When choosing a folder name without the `+` sign, everything works fine.

**Comment by raphywink** [comment]
Ok... still strange :-) but I am glad you found the culprit! Will leave the issue open as I still want to see what the `+` is exactly messing up.

**Comment by raphywink** [comment]
More of a note to self:

```r
# rename emuR_demoData/TextGrid_collection to bla+bli then
convert_TextGridCollection(dir = "~/Desktop/emuR_demoData/bla+bli/", dbName = "bla", targetDir = "~/Desktop/")
```
this reproduces the issue

**Comment by kirbyj** [comment]
> In other words if your have following folder structure:
> 
> * dir/subdir1/subsubdir1
> * dir/subdir2
> * dir
> 
> and every folder contains some TextGrid & wav file pair, `convert_TextGridCollection()` will preserve that folder structure in it's session structure:
> 
> * `0000_ses` containing the dir file pairs
> * `subdir1_ses` containig the subdir1 file pairs
> * `subdir1_subsubdir1_ses` ...
> * `subdir2_ses` ...

Is this why it isn't possible to assign arbitrary session names (e.g. "3" instead of "0000")?
